src/attention/utils/helpers.ts

```diff
@@ -1,98 +1,243 @@
+  actualDirection?: Directions
+  directionName?: Directions
+  arrowEl?: HTMLElement | null
+  attentionEl?: HTMLElement | null
+  flip?: Boolean
+  fallbackPlacements?: Directions[]
+  targetEl?: unknown
+  tooltip?: Boolean
+  popover?: Boolean
+  callout?: Boolean
+  highlight?: Boolean
+  noArrow?: Boolean
+  distance?: number
+  skidding?: number
+  waitForDOM?: () => void
+}
+
+const middlePosition = 'calc(50% - 7px)'
+const isDirectionVertical = (name: string) =>
+  [TOPSTART, TOP, TOPEND, BOTTOMSTART, BOTTOM, BOTTOMEND].includes(name)
+
 function computeCalloutArrow({
   actualDirection,
-  directionName,
+  directionName = 'bottom',
   arrowEl,
 }: AttentionState) {
-  if (!arrowEl) return;
+  if (!arrowEl) return
+
+  actualDirection = directionName
 
-  actualDirection = directionName;
-  const directionIsVertical = isDirectionVertical(directionName);
-  arrowEl.style.left = directionIsVertical ? middlePosition : "";
-  arrowEl.style.top = !directionIsVertical ? middlePosition : "";
+  const arrowDirection = opposites[actualDirection]
+  const arrowRotation = rotation[arrowDirection]
+
+  const directionIsVertical = isDirectionVertical(directionName)
+  arrowEl.style.left = directionIsVertical ? middlePosition : ''
+  arrowEl.style.top = !directionIsVertical ? middlePosition : ''
+  arrowEl.style.transform = `rotate(${arrowRotation}deg)`
 }
 
 export async function useRecompute(state: AttentionState) {
-  if (!state.isShowing) return; // we're not currently showing the element, no reason to recompute
-  await state?.waitForDOM?.(); // wait for DOM to settle before computing
-  if (state.isCallout) return computeCalloutArrow(state); // we don't move the callout box, only its arrow
-  const position = await computePosition(
-    state.targetEl as ReferenceElement,
-    state.attentionEl as HTMLElement,
-    {
-      placement: state.directionName,
-      middleware: [
-        // Should we make this configurable, but have these as sane defaults?
-        flip(),
-        offset(8),
-        shift({ padding: 16 }),
-        // @ts-ignore
-        arrow({ element: state.noArrow ? undefined : state.arrowEl }), // FIXME
-      ],
-    }
-  );
-  // @ts-ignore
-  state.actualDirection = position.placement;
-  Object.assign(state.attentionEl?.style || {}, {
-    left: "0",
-    top: "0",
-    transform: `translate3d(${Math.round(position.x)}px, ${Math.round(
-      position.y
-    )}px, 0)`,
-  });
-  // @ts-ignore
-  let { x, y } = position.middlewareData.arrow;
-
-  if (state.arrowEl) {
-    state.arrowEl.style.left = x ? x + "px" : "";
-    state.arrowEl.style.top = y ? y + "px" : "";
+  if (!state?.isShowing) return // we're not currently showing the element, no reason to recompute
+  if (state?.waitForDOM) {
+    await state?.waitForDOM() // wait for DOM to settle before computing
   }
+  if (state?.isCallout) return computeCalloutArrow(state)
+  
+  const referenceEl: ReferenceElement = state?.targetEl as ReferenceElement
```

Comment:
Do we need to write `: ReferenceElement` if it's already defined with `as ReferenceElement`?

Reply:
I am not sure how to remove the type assertion for `targetEl`, since that one has type `unknown` in, for example, our react-repo, and if I need to set it already in the react-repo to type `ReferenceElement` I would need to add `floating-ui/dom` devDependency to that project just for that. Type `ReferenceElement` is defined in `floating-ui/dom`. 

A potential solution could be to set the type of `targetEl` to `ReferenceElement` here in the core-repo that is part of the `AttentionState` type, and in the react-repo I could set the type of `targetEl` to what `floating-ui/dom` has defined `ReferenceElement` as, which is of type: `Element | VirtualElement` where `VirtualElement` is defined to this type: `{getBoundingClientRect: () => ClientRect | DOMRect, contextElement?: Element};`. But I'm not sure if we want to do this, or if it is just easier to keep it as it is now with the type assertion `as ReferenceElement`. Wdyt? :)

Reply:
However after thinking on it, maybe it's better if targetEl is of type `ReferenceElement | null` in the `AttentionState` type, and then I would just have to change the type of `targetEl` in our react-repo to also be of type: `ReferenceElement | null` and where I would have to create my own `ReferenceElement` type since that one is defined by `floating-ui/dom` and I don't think we want to add `floating-ui/dom` as a devDependency to the react-repo just to be able to use their `ReferenceElement` type. Wdyt? :)

Reply:
Updated in commit: [6c52c9d](https://github.com/warp-ds/core/pull/15/commits/6c52c9dd87f76c84b60c6c03389dbcd4302f36c2)